pkgs/build-support/fetchgit/tests.nix

```diff
@@ -79,4 +79,61 @@
     rev = "v3.0.14";
     sha256 = "sha256-bd0Lx75Gd1pcBJtwz5WGki7XoYSpqhinCT3a77wpY2c=";
   };
+
+  ssh-commit-verification = testers.invalidateFetcherByDrvHash fetchgit {
+    name = "ssh-commit-verification-source";
+    url = "https://codeberg.org/flandweber/git-verify";
+    rev = "a43858e8f106b313aed68b6455a45340db7dd758";
+    sha256 = "sha256-tryIB8KlETlbHyTfW+IpsAgu2BQCcoeuHJpvzyFFsMg=";
+    verifyCommit = true;
+    publicKeys = [{
+        type = "ssh-ed25519";
+        key = "AAAAC3NzaC1lZDI1NTE5AAAAIBiNDWMPZNRItkm1U1CQkJUUrrmM+l7CdE6wyUHzr4Nr";
+      }];
+  };
+
+  gpg-commit-verification = testers.invalidateFetcherByDrvHash fetchgit {
+    name = "gpg-commit-verification-source";
+    url = "https://gitlab.torproject.org/tpo/core/tor";
+    rev = "8888e4ca6b44bb7476139be4644e739035441b35";
+    sha256 = "sha256-mC18zGHAh/u9zcYMoYjgrQfsOtjp0UwfP4JXGZWzJHs=";
+    verifyCommit = true;
+    publicKeys = [{
+        type="gpg";
+        key= fetchurl {
+          url = "https://keys.openpgp.org/vks/v1/by-fingerprint/5EF3A41171BB77E6110ED2D01F3D03348DB1A3E2";
+          sha256 = "sha256-xvBWfaS1py7vyDIIYGtATqBOnWafd3B6OB2Blhfm4MU=";
```

Comment:
I think the contents of this URL will change if the key owner edits the key in any way (e.g. modifies expiry dates or adds a subkey) and then uploads their new key. We could use archive.org to keep an immutable version of this key but maybe there is a better solution.

Reply:
makes sense, thanks!